### PR TITLE
Some Relion 3.1 fixes

### PR DESCRIPTION
**Pull request by delarosatrevin**
Bug fixes and features:

- [x] * Fixed matching between particles and movies by using micName in polishing
- [x] * Fixed generation of the local motion table when needed (e.g when only some movies failed to align)
- [x] * Allowed to display ctf-refinement results with matplotlib (still need to find the proper color map)
- [x] * Fixed function splitCTFGroups, now replaced with class DefocusGroups and not using Xmipp metadata 
- [x] * Now use proper pixelSize label depending on input set

### CODE REVIEW

File: relion/viewers/viewer_ctfrefine.py [hunk 1]
```diff
@@ -170,23 +175,47 @@ def _displayDefocus(self, e=None):
...
 
+    def _showImages(self, *imgs, **kwargs):
+        title = kwargs.get('title', '')
+        if self.useMatplotlib:
+            return self._showImagesMatplotlib(title, *imgs)
+        else:
+            return [DataView(img) for img in imgs]
+
+    def _showImagesMatplotlib(self, title, *imgs):
+        import mrcfile
```
Comment (azazellochg) [inline relion/viewers/viewer_ctfrefine.py:186]: for this we'll need to include it into pip requirements?

Comment (delarosatrevin) [reply relion/viewers/viewer_ctfrefine.py:186]: I wasn't sure if we were already using mrcfile somewhere...we could also read the image with ImageHandler class (xmipp under the hood) and use the getData() that will give us the numpy array...probably the best solution right now....anyway this import should not stay there.

**Comment by delarosatrevin** [comment]
@azazellochg Can you take a look at the PR?

**Comment by azazellochg** [comment]
Looking now

File: relion/viewers/viewer_ctfrefine.py (hunk 1 above)
Comment (azazellochg) [reply relion/viewers/viewer_ctfrefine.py:186]: it's still there.. :)

Comment (delarosatrevin) [reply relion/viewers/viewer_ctfrefine.py:186]: Yes, I will change this, please keep checking and I will fix this before the merge.

File: relion/protocols/protocol_base.py [hunk 2]
```diff
@@ -30,6 +30,7 @@
 from collections import OrderedDict
 from emtable import Table
 
+from pyworkflow.object import String
```
Comment (azazellochg) [inline relion/protocols/protocol_base.py:33]: Didn't find where this is used

File: relion/convert/convert_coordinates.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
+# **************************************************************************
```
Comment (azazellochg) [inline relion/convert/convert_coordinates.py:1]: Am I right that this is just moved code? Nothing new?

File: relion/protocols/protocol_base.py [hunk 4]
```diff
@@ -991,7 +993,14 @@ def _setComputeArgs(self, args):
         args['--pool'] = self.pooledParticles.get()
 
         if self.doGpu:
-            args['--gpu'] = self.gpusToUse.get()
+            gpuStr = self.gpusToUse.get().strip()
```
Comment (azazellochg) [inline relion/protocols/protocol_base.py:996]: I didn't have any problems running relion without quotes, but why not

**Comment by azazellochg** [comment]
@delarosatrevin I have run ref3d with grouping by defocus. Input star has groups, but output particles set does not have rlnGroupName or number

File: relion/protocols/protocol_base.py (hunk 4 above)
Comment (delarosatrevin) [reply relion/protocols/protocol_base.py:996]: I can't remember exactly the case, but Dustin had some issue, not  sure if using :  or spaces

File: relion/convert/convert_coordinates.py (hunk 3 above)
Comment (delarosatrevin) [reply relion/convert/convert_coordinates.py:1]: Yes, moved code from convert_deprecated mainly...there were a lot of coordinates related functions that might help to have in its own implementation file.

File: relion/viewers/viewer_ctfrefine.py (hunk 1 above)
Comment (delarosatrevin) [reply relion/viewers/viewer_ctfrefine.py]: Done now with ImageHandler

File: relion/protocols/protocol_base.py (hunk 2 above)
Comment (delarosatrevin) [reply relion/protocols/protocol_base.py]: Removed

**Comment by delarosatrevin** [comment]
@azazellochg I have addressed your comments, still, the reconstruct test is failing, but I will suggest leaving out of this PR. I asked Marabini help for that.

**Review by azazellochg — APPROVED** [review]

**Pull request merged**